src/Tools/dotnet-stack/Symbolicate.cs

```diff
@@ -0,0 +1,419 @@
+    {
+        // Temporary folder to store the files converted from pdb to xml
+        private static readonly string tempDirectory = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+
+        delegate void SymbolicateDelegate(FileInfo[] inputPath, string searchDir, string output);
+
+        /// <summary>
+        /// Get the line number from the Method Token and IL Offset at the stacktrace
+        /// </summary>
+        /// <param name="inputPath">The input path for file with stacktrace text</param>
+        /// <param name="searchDir">All paths in the directory to the assembly and pdb where the exception occurred</param>
+        /// <param name="output">The output path for the extracted line number data</param>
+        /// <returns></returns>
+        private static void Symbolicate(FileInfo[] inputPath, string searchDir, string output)
+        {
+            try
+            {
+                if (inputPath == null)
+                {
+                    throw new InvalidDataException("Required argument missing for command\n");
+                }
+
+                string input = string.Empty;
+                if (inputPath.Length > 1)
+                {
+                    for (int i = 0; i < inputPath.Length; i++)
+                    {
+                        Console.WriteLine($" {i + 1}. {inputPath[i]}");
+                    }
+                    Console.Write($"Select one of several stacktrace files [1-{inputPath.Length}]: ");
+                    input = inputPath[Int32.Parse(Console.ReadLine()) - 1].FullName;
+                }
+                else
+                {
+                    input = inputPath[0].FullName;
+                }
+                if (!File.Exists(input))
+                {
+                    throw new FileNotFoundException($"{input} file does not exist\n");
+                }
+
+                string[] search_paths = null;
+                if (searchDir != null)
+                {
+                    search_paths = searchDir.Split(";");
+                }
+                search_paths ??= new string[] { Directory.GetCurrentDirectory() };
+
+                Symbolicator(PdbToXmlConvert(search_paths), input, output);
+            }
+            catch (Exception e)
+            {
+                Console.Error.WriteLine(e.Message);
+            }
+        }
+
+        private static List<string> PdbToXmlConvert(string[] searchPaths)
+        {
+            List<string> peFiles = GrabFiles(searchPaths, "*.dll");
+            if (peFiles.Count == 0)
+            {
+                throw new FileNotFoundException("Assembly file not found\n");
+            }
+
+            List<string> pdbFiles = GrabFiles(searchPaths, "*.pdb");
+            if (pdbFiles.Count == 0)
+            {
+                throw new FileNotFoundException("PDB file not found\n");
+            }
+
+            Console.Out.Write("Converting pdb to xml...");
+            Console.SetCursorPosition(0, Console.CursorTop);
```

Comment:
This will throw an exception under redirection. The logging should either check if input/output is redirected or no attempt to rewrite console lines.

Reply:
I'll delete it. Because it's unnecessary log.

Reply:
I modified it.